lib/Doctrine/ORM/PersistentCollection.php

```diff
@@ -514,10 +514,10 @@ public function offsetGet($offset)
     public function offsetSet($offset, $value)
     {
         if ( ! isset($offset)) {
-            return $this->add($value);
+            $this->add($value);
```

Comment:
I'm not sure about this one since now both `add()` and `set()` will be called if `isset($offset) === false`, so we should at least have a `return` after `$this->add($value);`